script/release-notes/notes.js

```diff
@@ -0,0 +1,553 @@
+
+  return note
+}
+
+/**
+ * Looks for our project's conventions in the commit message:
+ *
+ * 'semantic: some description' -- sets type, subject
+ * 'some description (#99999)' -- sets subject, pr
+ * 'Fixes #3333' -- sets issueNumber
+ * 'Merge pull request #99999 from ${branchname}' -- sets pr
+ * 'This reverts commit ${sha}' -- sets revertHash
+ * line starting with 'BREAKING CHANGE' in body -- sets breakingChange
+ * 'Backport of #99999' -- sets pr
+ */
+const parseCommitMessage = (commitMessage, owner, repo, o = {}) => {
+  // console.log('parse receiving', { o })
+
+  // split commitMessage into subject & body
+  let subject = commitMessage
+  let body = ''
+  const pos = subject.indexOf('\n')
+  if (pos !== -1) {
+    body = subject.slice(pos).trim()
+    subject = subject.slice(0, pos).trim()
+  }
+
+  if (!o.originalSubject) o.originalSubject = subject
+
+  if (body) {
+    o.body = body
+
+    const note = getNoteFromBody(body)
+    if (note) { o.note = note }
+  }
+
+  // if the subject ends in ' (#dddd)', treat it as a pull request id
+  let match
+  if ((match = subject.match(/^(.*)\s\(#(\d+)\)$/))) {
+    setPullRequest(o, owner, repo, parseInt(match[2]))
+    subject = match[1]
+  }
+
+  // if the subject begins with 'word:', treat it as a semantic commit
+  if ((match = subject.match(/^(\w+):\s(.*)$/))) {
+    o.type = match[1].toLocaleLowerCase()
+    subject = match[2]
+  }
+
+  // Check for GitHub commit message that indicates a PR
+  if ((match = subject.match(/^Merge pull request #(\d+) from (.*)$/))) {
+    setPullRequest(o, owner, repo, parseInt(match[1]))
+    o.pr.branch = match[2].trim()
+  }
+
+  // Check for a trop comment that indicates a PR
+  if ((match = commitMessage.match(/\bBackport of #(\d+)\b/))) {
+    setPullRequest(o, owner, repo, parseInt(match[1]))
+  }
+
+  // https://help.github.com/articles/closing-issues-using-keywords/
+  if ((match = subject.match(/\b(?:close|closes|closed|fix|fixes|fixed|resolve|resolves|resolved|for)\s#(\d+)\b/))) {
+    o.issueNumber = parseInt(match[1])
+    if (!o.type) o.type = 'fix'
+  }
+
+  // look for 'fixes' in markdown; e.g. 'Fixes [#8952](https://github.com/electron/electron/issues/8952)'
+  if (!o.issueNumber && ((match = commitMessage.match(/Fixes \[#(\d+)\]\(https:\/\/github.com\/(\w+)\/(\w+)\/issues\/(\d+)\)/)))) {
+    o.issueNumber = parseInt(match[1])
+    if (o.pr && o.pr.number === o.issueNumber) o.pr = null
+    if (o.originalPr && o.originalPr.number === o.issueNumber) o.originalPr = null
+    if (!o.type) o.type = 'fix'
+  }
+
+  // https://www.conventionalcommits.org/en
+  if (body.split('\n').map(x => x.trim()).some(x => x.startsWith('BREAKING CHANGE'))) {
+    o.type = 'breaking-change'
+  }
+
+  // Check for a reversion commit
+  if ((match = body.match(/This reverts commit ([a-f0-9]{40})\./))) {
+    o.revertHash = match[1]
+  }
+
+  // Edge case: manual backport where commit has `owner/repo#pull` notation
+  if (commitMessage.includes('ackport') &&
```

Comment:
I think its better to be more precise here with:
```
if (commitMessage.toLowerCase().includes('backport') &&
```
It also helps if someone is looking for the word `backport` in the code.

Reply:
Fixed 037831a5d13ab78baca2bb10b227cb4b3a6b18a3